gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/core/api/use_case/UpdateFederatedApiUseCaseTest.java

```diff
@@ -148,28 +174,101 @@ public void update_federation_api_with_basic_configuration_info() {
         //given
         apiCrudService.initWith(List.of(ApiFixtures.aFederatedApi()));
         var auditInfo = AuditInfoFixtures.anAuditInfo(ORGANIZATION_ID, ENVIRONMENT_ID, USER_ID);
+        String categoryKey = "categoryKey-1";
         var apiToUpdate = ApiFixtures
             .aFederatedApi()
             .toBuilder()
             .name("updated-name")
             .description("updated-description")
             .version("2.0.0")
             .labels(List.of("label-1"))
-            .categories(Set.of("category-1"))
+            .categories(Set.of(categoryKey))
             .apiLifecycleState(Api.ApiLifecycleState.PUBLISHED)
             .build();
 
+        CategoryEntity categoryEntity = new CategoryEntity();
+        String categoryId = "categoryId-1";
+        categoryEntity.setId(categoryId);
+        when(categoryService.findById(categoryKey, ENVIRONMENT_ID)).thenReturn(categoryEntity);
```

Comment:
In these tests, we don't want to mock services at all. The advantage is that we can have more thorough end to end tests. Once you have the service within the onion, you will need to initialize data in the CategoryQueryServiceInMemory I believe, and then everything should work.

Reply:
Resolved 1-1